enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -572,6 +573,8 @@ func (l *FileCacheLoader) UnpackSnapshot(ctx context.Context, snapshot *Snapshot
 }
 
 func (l *FileCacheLoader) CacheSnapshot(ctx context.Context, key *fcpb.SnapshotKey, opts *CacheSnapshotOptions) error {
+	key = generateMergeQueueSupportedKey(key)
```

Comment:
Why is it safe to only mutate this key inside of this one method?

Reply:
Let me know if this description helps explain the logic

Incoming workflows triggered by webhook will automatically have fallback_branch=main

So a PR workflow might have { branch=myb, fallback=main }
A merge queue workflow might have { branch = mergeq, fallback=main }

1. With this PR, the merge queue workflow will save a snapshot to ref=main
2. An incoming PR workflow will not find a snapshot for ref=myb, so will fallback to ref=main (it will pick the snapshot from step #1) 
3. An incoming merge queue workflow will not find a snapshot for ref=mergeq, so will fall back to ref=main (it will pick the snapshot from step #1). It will then overwrite the snapshot at ref=main. This is good, as it represents the latest state of the merge queue, which should soon represent main (as long as CI plasses)

Reply:
wdyt about asking the customer(s) to run CI on main instead? it addresses the conditional part of "as long as CI passes" because main should always be green, and it shouldn't add much to their costs because the build should be fully cached?

Reply:
(asking because the branching model for snapshotting is already pretty confusing, so I think the more we can do to reduce the amount of edge cases to consider, the better)

Reply:
Given that merge queues are a fairly common use case, I feel like we should support it out of the box and without requiring customers to pay more for it. I could see a user cleaning up the workflow on main (because it functionally doesn't do anything), and that unexpectedly breaking snapshot sharing, which is also unexpected. What do you think?

Reply:
I think overall, saving the snapshot to the "main" key seems fine. To Tyler's point, I wonder if this can be modeled in a more direct way - we have SnapshotKeySet which holds all the keys associated with the snapshot - could we maybe store some sort of "target" key there (i.e. the key that will be overwritten) and then pass it directly to this func?

Reply:
Updated to be more explicit